skimage/measure/_moments.py

```diff
@@ -345,7 +345,8 @@ def moments_hu(nu):
     array([7.45370370e-01, 3.51165981e-01, 1.04049179e-01, 4.06442107e-02,
            2.64312299e-03, 2.40854582e-02, 4.33680869e-19])
     """
-    return _moments_cy.moments_hu(nu.astype(np.double))
+    dtype = np.float32 if nu.dtype == 'float32' else np.float64
```

Comment:
This seems fine, but I wonder if we should explicitly raise on error on complex-valued inputs? I think there are probably a ton of other places where this is just assumed and not explicitly checked, thgouh. not sure it is worth it

```
if nu.dtype.kind = 'c':
    raise TValueError("complex-valued inputs are not suppoted.")
```

Reply:
I have no strict opinion on this, but most of skimage functions do not explicitly check for complex input values, I am not sure it worth it :confused: